conference/talk_voting.py

```diff
@@ -65,13 +65,25 @@ def is_user_allowed_to_vote(user):
     This usually means checking if they have at least one ticket associated
     with their account (either for this or any of the past years
     """
-    return user.ticket_set.all().exists()
+    is_allowed = (
+        user.ticket_set.all().exists()
+        or Talk.objects.proposed().filter(created_by=user, conference=Conference.objects.current().code).exists()
+    )
+    return is_allowed
 
 
 @login_required
 def vote_on_a_talk(request, talk_uuid):
     talk = get_object_or_404(Talk, uuid=talk_uuid)
 
+    # Users can't vote on their own talks.
+    if talk.speakers.filter(pk=request.user.pk).exists():
```

Comment:
I guess here we don't need to return the voting form at all, just return 200, (or 403?), because on the client we wouldn't show the buttons either so the UI shouldn't break from providing a different response.

Reply:
Not sure I understand - what I was going for was preserving the same response here as if `No Vote` was submitted, in case we have a bug and do show the vote buttons.

Reply:
Makes sense, and it's equally valid, just thought that we could simplify to 403 if we're not particularly interested in the buttons in this case.